addons/website_sale/models/website_snippet_filter.py

```diff
@@ -61,6 +63,8 @@ def _filter_records_to_values(self, records, is_sample=False):
                     res_product.update(product._get_combination_info_variant())
                     if records.env.context.get('add2cart_rerender'):
                         res_product['_add2cart_rerender'] = True
+                    if not self.env.context.get('show_variants'):
+                        product.display_name = product.name
```

Comment:
You can't change a record on the fly like this. 

If `show_variants` is `false`, we want to show product template records. You only change the name but the record can also have a different product price, picture etc. Please adapt the code to use `product.template` directly.